django/db/models/fields/__init__.py

```diff
@@ -898,110 +898,6 @@ def value_from_object(self, obj):
         return getattr(obj, self.attname)
 
 
-class AutoField(Field):
-    description = _("Integer")
-
-    empty_strings_allowed = False
-    default_error_messages = {
-        'invalid': _('“%(value)s” value must be an integer.'),
-    }
-
-    def __init__(self, *args, **kwargs):
-        kwargs['blank'] = True
-        super().__init__(*args, **kwargs)
-
-    def check(self, **kwargs):
-        return [
-            *super().check(**kwargs),
-            *self._check_primary_key(),
-        ]
-
-    def _check_primary_key(self):
-        if not self.primary_key:
-            return [
-                checks.Error(
-                    'AutoFields must set primary_key=True.',
-                    obj=self,
-                    id='fields.E100',
-                ),
-            ]
-        else:
-            return []
-
-    def deconstruct(self):
-        name, path, args, kwargs = super().deconstruct()
-        del kwargs['blank']
-        kwargs['primary_key'] = True
-        return name, path, args, kwargs
-
-    def get_internal_type(self):
-        return "AutoField"
-
-    def to_python(self, value):
-        if value is None:
-            return value
-        try:
-            return int(value)
-        except (TypeError, ValueError):
-            raise exceptions.ValidationError(
-                self.error_messages['invalid'],
-                code='invalid',
-                params={'value': value},
-            )
-
-    def rel_db_type(self, connection):
-        return IntegerField().db_type(connection=connection)
-
-    def validate(self, value, model_instance):
-        pass
```

Comment:
`AutoFieldMixin.validate()`  is missing , I think we should restore `AutoField.validate()` implementation, i.e.
```python
def validate(self, value, model_instance):
    pass
```

Reply:
Interesting. It has been like this since 471596fc1afcb9c6258d317c619eaf5fd394e797, but I don't really understand the reason that it is bypassing [`Field.validate()`](471596fc1afcb9c6258d317c619eaf5fd394e797), unless it is a minor optimisation to skip it because none of those checks really make sense for the auto fields...

Will restore this.